NewareNDA/NewareNDA.py

```diff
@@ -27,15 +29,30 @@ def read(file, software_cycle_number=True, cycle_mode='chg'):
             'chg': (Default) Sets new cycles with a Charge step following a Discharge.
             'dchg': Sets new cycles with a Discharge step following a Charge.
             'auto': Identifies the first non-rest state as the incremental state.
+        log_level (str): Sets the modules logging level. Default: 'INFO'
+            Options: 'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'
     Returns:
         df (pd.DataFrame): DataFrame containing all records in the file
     """
+    
+    
+    # Set up logging
+    log_level = log_level.upper()
+    if log_level in logging._nameToLevel.keys():
+        logger.setLevel(log_level)
+    else:
+        warnings.warn(f"Logging level '{log_level}' not supported; Defaulting to 'INFO'. "
+                      "Supported options are: {', '.join(logging._nameToLevel.keys())}")
+        logger.warn(f"Logging level '{log_level}' not supported; Defaulting to 'INFO'.")

```

Comment:
I think it is superfluous having both warnings.warn and logger.warn. Default log level is info so both messages will be printed, currently I get something like (after fixing message as above):

```
NewareNDA.py:44: UserWarning: Logging level 'FOO' not supported; Defaulting to 'INFO'. Supported options are: {CRITICAL, FATAL, ERROR, WARN, WARNING, INFO, DEBUG, NOTSET}
  warnings.warn(f"Logging level '{log_level}' not supported; Defaulting to 'INFO'. "
Logging level 'FOO' not supported; Defaulting to 'INFO'.
```

when running from terminal.

Reply:
I see, so `logger.warning()` DOES print to the console. This is not stated in the logging docs under this command! I suppose it is implicit to the basicConfig?
In that case, I agree: the extra `warnings.warn()` call is not needed.

Reply:
Indeed, from logging docs it seems like basicConfig is needed to redirect output to file (or another stream)